core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala

```diff
@@ -147,6 +147,18 @@ class SparkThrowableSuite extends SparkFunSuite {
     assert(rereadErrorClassToInfoMap == errorReader.errorInfoMap)
   }
 
+  test("Error class names should contain only capital letters, numbers and underscores") {
+    val allowedChars = "[A-Z0-9_]*"
+    errorReader.errorInfoMap.foreach { e =>
+      assert(e._1.matches(allowedChars))
```

Comment:
This one doesn't show which error class among 1500 is invalid, see:
```
e._1.matches(allowedChars) was false
ScalaTestFailureLocation: org.apache.spark.SparkThrowableSuite at (SparkThrowableSuite.scala:153)
org.scalatest.exceptions.TestFailedException: e._1.matches(allowedChars) was false
```
Could you output the error class name:
```suggestion
      assert(e._1.matches(allowedChars), s"The error class is invalid: ${e._1}")
```

Reply:
Thanks for the suggestion. Updated.